### PR TITLE
fix: zIndex miss of message

### LINKED ISSUE #45883: Nested modal with notification is not correctly being z-Indexed
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/modal-notification-antd-5-11-1-forked-xnrk2r?file=/content2.tsx)

### Steps to reproduce

Click on the button to open the first modal, then click on the button to open the second modal, then click on the button to fire the notification

### What is expected?

The notification should appear above all the modals and masks, correctly being z-indexed as the top element of the page

### What is actually happening?

The notification is being shown behind the later modal mask, and this way it can't be closed or interacted with, and also its harder to see depending of your mask styling.

| Environment | Info |
| --- | --- |
| antd | 5.11.1 |
| React | 18.0.0 |
| System | Windows 11 |
| Browser | Brave |

---
I tried changing the notification zIndex itself but it does not result in any change at all, also tried changing the modal masks but it didnt work as before either. This issue started happening on recent versions of Antd, but i dont know which one exactly, since i only detected it today, so perhaps some internal change can maybe have caused this? On my project, I didnt change anything in my code who handles this notifications for more than 6 months.

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fixes #45883 #45912

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Optimize `zIndex` logic, fix the problem that message and notification are covered when multiple Modal are opened.       |
| 🇨🇳 Chinese |     优化 `zIndex` 逻辑，修复多层 Modal 打开时，message 与 notification 被遮盖的问题。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at dd5413f</samp>

This pull request improves the z-index management and consistency of various components, such as `modal`, `message`, `notification`, and `header`. It also fixes the issue of the header slot overlapping with the modal close button, and updates the documentation for the new version 5.11.1.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at dd5413f</samp>

* Reduce the z-index of the header slot to avoid overlapping with other components ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-2b1224dbeb7d20f1cb1b4590eb12b68104c605d2a67f0b0bd4fd9d8b0100a728L47-R47))
* Refactor the z-index offset constants for container components to use a common base value and a multiplier ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-2a902ec0f175b29920c300ea9a4023433a9a9720749632f5ac132e73694b1f71L10-R26))
* Modify the z-index calculation for container components to use the preset token value by default, not stack it when nested, and cap it by a maximum value ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-2a902ec0f175b29920c300ea9a4023433a9a9720749632f5ac132e73694b1f71L38-R54))
* Increase the z-index value for the message and notification components by adding the maximum offset to the base value ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL198-R200), [link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L258-R260))
* Update the import order and add the missing import of the `CONTAINER_MAX_OFFSET` constant in the style files of the message and notification components ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-8af0be282a46c48f41d0fd430f9b0f81669ae42e3e93d0da8dfd635c6f24b7dfL2-R6), [link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-1b72cd8f976d21d83517b2c963a0b9a9bb6670e5d004ed4a7b0dd2edad3fc965L3-R6))
* Change the name of the nested modal demo component from `App` to `Demo` and call the `useMessage` hook to create an instance and a holder for the message component ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-740d231aa0062e7ab64fb1c11b1a82f487d219f392ddcbf2be01b45a8538e1e6L15-R16))
* Extend the nested modal demo to include a button that triggers a message component and render the holder element after the modal components ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-740d231aa0062e7ab64fb1c11b1a82f487d219f392ddcbf2be01b45a8538e1e6L82-R99))
* Update the post-script script to include the version 5.11.1 and the link to the issue that was fixed in this version ([link](https://github.com/ant-design/ant-design/pull/45911/files?diff=unified&w=0#diff-f90e598624ba13764610410ae49732147dec4738e55891b356f6162482dc8923R52))

### CODE REVIEW

File: components/notification/style/index.ts [hunk 1]
```diff
@@ -255,7 +257,7 @@ const genNotificationStyle: GenerateStyle<NotificationToken> = (token) => {
 
 // ============================== Export ==============================
 export const prepareComponentToken = (token: AliasToken) => ({
-  zIndexPopup: token.zIndexPopupBase + 50,
+  zIndexPopup: token.zIndexPopupBase + CONTAINER_MAX_OFFSET + 50,
```
Comment (li-jia-nan) [inline components/notification/style/index.ts:260]: 感觉会跟 feature 有冲突

File: .dumi/theme/slots/Header/index.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const useStyle = createStyles(({ token, css }) => {
     header: css`
       position: sticky;
       top: 0;
-      z-index: 2000;
+      z-index: 1000;
```
Comment (kiner-tang) [inline .dumi/theme/slots/Header/index.tsx:47]: 这个我记得昨天刚有个 pr 改的

File: components/notification/style/index.ts (hunk 1 above)
Comment (kiner-tang) [reply components/notification/style/index.ts:260]: 应该不会，cssVar 好像没改到 zIndex

**Review by kiner-tang — APPROVED** [review]

**Comment by kiner-tang** [comment]
好多镜像都要更新一下

File: .dumi/theme/slots/Header/index.tsx (hunk 2 above)
Comment (zombieJ) [reply .dumi/theme/slots/Header/index.tsx:47]: 嗯，之前那个 PR 不对。Popover 本来就是应该是在最上面的：

image

**Comment by kiner-tang** [comment]
感觉得全量更新一下快照，Popover 用的地方好像有点多

**Pull request merged**